.storybook/stories/TableFiltering/components/DashboardTable.tsx

```diff
@@ -0,0 +1,262 @@
+import { TextLink } from '@ag.ds-next/react/text-link';
+import { Text } from '@ag.ds-next/react/text';
+import { Fragment } from 'react';
+import { format } from 'date-fns';
+import {
+	BusinessForAudit,
+	BusinessForAuditWithIndex,
+} from '../lib/generateBusinessData';
+import { Avatar } from '@ag.ds-next/react/avatar';
+import { Flex } from '@ag.ds-next/react/box';
+import { GetDataSort } from '../lib/getData';
+import {
+	DashboardTableHeader,
+	DashboardTableHeaderBaseProps,
+} from './DashboardTableHeader';
+
+const DashboardTableRowAssignee = ({
+	assignee,
+}: {
+	/** The name of the assignee, if set */
+	assignee: string | undefined;
+}) => {
+	if (!assignee) {
+		return <TableCell>-</TableCell>;
+	}
+
+	return (
+		<TableCell>
+			<Flex alignItems="center" gap={0.25}>
+				<Avatar name={assignee} size="sm" />
+				<Text>{assignee}</Text>
+			</Flex>
+		</TableCell>
+	);
+};
+
+export type DashboardTableProps = {
+	/** The caption for the table */
+	caption: string;
+	/** The data to display in the table */
+	data: BusinessForAuditWithIndex[];
+	/** The current sort */
+	sort?: GetDataSort;
+	/** A function to set the sort */
+	setSort?: (sort: GetDataSort) => void;
+	/** Whether the table is loading */
+	loading?: boolean;
+	/** The total number of items found in the search */
+	totalItems?: number;
+	/** The number of items to display per page */
+	itemsPerPage?: number;
+};
+
+export const DashboardTable = ({
+	caption,
+	data,
+	loading,
+	sort,
+	setSort,
+	itemsPerPage = 10,
+	totalItems,
+}: DashboardTableProps) => {
+	const isTableSortable = !sort || !!setSort;
+
+	if (!loading && data.length === 0) {
+		return <Text>No results</Text>;
+	}
+
+	return (
+		<TableWrapper>
+			<Table aria-rowcount={totalItems}>
+				<TableCaption>
+					{caption}
+					<VisuallyHidden>
+						, column headers with buttons are sortable.
+					</VisuallyHidden>
+				</TableCaption>
+				<TableHead>
+					<tr>
+						{headers.map(
+							({
+								label,
+								sortKey,
+								textAlign,
+								width,
+								isSortable: isFieldSortable,
+							}) => {
+								const isFieldTheActiveSortField = sort?.field === sortKey;
+								const clickHandler = () =>
+									setSort?.({
+										field: sortKey,
+										order:
+											sort?.field === sortKey && sort?.order === 'ASC'
+												? 'DESC'
+												: 'ASC',
+									});
+
+								return (
+									<DashboardTableHeader
+										key={sortKey}
+										textAlign={textAlign}
+										width={width}
+										isSortable={isTableSortable && isFieldSortable}
+										sort={isFieldTheActiveSortField ? sort.order : undefined}
+										onClick={clickHandler}
+									>
+										{label}
+									</DashboardTableHeader>
+								);
+							}
+						)}
+					</tr>
+				</TableHead>
+				<TableBody>
+					{loading ? (
+						<Fragment>
+							{Array.from(Array(itemsPerPage).keys()).map((i) => (
+								<tr key={i}>
+									<TableCell>
```

Comment:
I'd highly recommend that [use this technique "to tame data tables"](https://csswizardry.com/2013/11/taming-data-tables/) so that the columns don't shift about between loading states 
![table-shifting](https://user-images.githubusercontent.com/853552/224211969-cbc2d285-639b-44a1-bb8f-21e78311d4d8.gif)

Reply:
Great advice Matt!